lib/synapse/config_generator/haproxy.rb

```diff
@@ -1005,7 +1028,8 @@ def generate_backend_stanza(watcher, config)
         log.debug "synapse: no backends found for watcher #{watcher.name}"
       end
 
-      keys = case watcher.haproxy['backend_order']
+      watcher_haproxy = watcher.generator_config['haproxy']
```

Comment:
whenever we do this, lets keep the name consistent. i propose `watcher_opts` as a variable that generally refers to the watcher-specific options for the current generator